
## simpleaf 
```bash
conda activate simpleaf

export AF_SAMPLE_DIR=your_directory
mkdir $AF_SAMPLE_DIR
cd $AF_SAMPLE_DIR
```

```bash 
# define env variable
export ALEVIN_FRY_HOME="$AF_SAMPLE_DIR/af_home"

# simpleaf configuration
simpleaf set-paths

# Increase the number of open files for piscem's intemediate files
ulimit -n 2048

# Define reference and index directories
IDX_DIR="your_index_directory"

# Ensure the index directory exists
mkdir -p $IDX_DIR

```

## create index
```bash
# Define the existing FASTA and GTF file paths
FASTA_FILE="$AF_SAMPLE_DIR/reference/all.ref.all.hap.fasta"
GTF_FILE="$AF_SAMPLE_DIR/reference/all.ref.all.hap.gtf"


# Run simpleaf index with existing FASTA and GTF
nohup simpleaf index \
--use-piscem \
--output "$IDX_DIR" \
--fasta "$FASTA_FILE" \
--gtf "$GTF_FILE" \
--rlen 150 \
--threads 8 \
--work-dir "$AF_SAMPLE_DIR/af_home/workdir.noindex" &

#check t2g mat


```

## define read 1 and 2 files
```bash

# Define filename pattern
reads1_pat="_R1_"
reads2_pat="_R2_"


reads1="$(find -L /DATA15/Projects/Spatial_transcriptomics_PM/10xChromium/Project_DCMB_10XSC0407/FASTQS/Sample2A -name "*$reads1_pat*" -type f | sort | awk -v OFS=, '{$1=$1;print}' | paste -sd, -)"

reads2="$(find -L /DATA15/Projects/Spatial_transcriptomics_PM/10xChromium/Project_DCMB_10XSC0407/FASTQS/Sample2A -name "*$reads2_pat*" -type f | sort | awk -v OFS=, '{$1=$1;print}' | paste -sd, -)"
```

## mapping
We will use alevin-fry because it is salmon-based.
The GTF file has to contain the gene_id and gene_name column.


```bash 

#if you have only one lane per sample:
for sample_dir in $(ls -d /your_sample_folder/Sample*); do sample_name=$(echo $sample_dir | sed 's:^.*/Sample:Sample:'); reads1=$(ls ${sample_dir}/*_R1_001.fastq.gz | tr '\n' ' ' | sed 's: $::g'); reads2=$(ls ${sample_dir}/*_R2_001.fastq.gz | tr '\n' ' ' | sed 's: $::g'); echo "simpleaf quant --reads1 $reads1 --reads2 $reads2 --threads 24 --index "$IDX_DIR/index" --unfiltered-pl /cellranger/barcodes/3M-3pgex-may-2023_TRU.txt.gz  --chemistry 10xv4-3p --resolution cr-like --anndata-out --output $AF_SAMPLE_DIR/${sample_name}_quant"; done > commands.sh


nohup parallel -j 2 :::: commands.sh &


#samples 1-6 (2 lanes)

for sample_dir in /your_sample_folder/Sample{1..6}; do
    sample_name=$(basename "$sample_dir")  # e.g. Sample1

    # Get all matching R1 and R2 files, comma-separated
    reads1=$(find "$sample_dir" -name "*_R1_001.fastq.gz" | sort | paste -sd "," -)
    reads2=$(find "$sample_dir" -name "*_R2_001.fastq.gz" | sort | paste -sd "," -)

    echo "simpleaf quant --reads1 \"$reads1\" --reads2 \"$reads2\" --threads 24 --index \"$IDX_DIR/index\" --unfiltered-pl \"/cellranger/barcodes/3M-3pgex-may-2023_TRU.txt.gz\" --chemistry 10xv4-3p --resolution cr-like --anndata-out --output \"$AF_SAMPLE_DIR/${sample_name}_quant\""
done > commands.sh

nohup parallel -j 2 :::: commands.sh &


```

## quality control
``` R
library(alevinQC)


###get mapping rate 

simpleafQCReport(
  simpleafQuantDir="./SampleX_quant",
  sampleId="SampleX",
  outputFile="alevin_report_$SampleX.html",
  outputDir = "./",
  outputFormat = "html_document",
  showCode = FALSE,
  forceOverwrite = FALSE,
  knitrProgress = FALSE,
  quiet = FALSE,
  ignorePandoc = FALSE,
  customCBList = list()
)


```
