
## obtain the list of defense related genes
Massonnet et al., 2022 - Supplemental Table 6
```R
### prep data  for synteny between PN40024 and Cabernet Sauvignon clone 08
# convert gff3 to bed  
python -m jcvi.formats.gff bed --type=mRNA --key=ID --primary_only ref.gff3 -o ref.bed  
python -m jcvi.formats.gff bed --type=mRNA --key=ID --primary_only query.gff3 -o query.bed  
  
# reformat the fasta  
python -m jcvi.formats.fasta format ref.fasta ref.cds  
python -m jcvi.formats.fasta format query.fasta query.cds  
  
### synteny analysis  
# pairwise synteny search using LAST  
python -m jcvi.compara.catalog ortholog ${query} ref --no_strip_names > ${query}.on.ref.log 2> ${query}.on.ref.err  
  
### microsynteny analysis  
# get the blocks file  
python -m jcvi.compara.synteny mcscan ref.bed ${query}.ref.lifted.anchors --iter=2 -o ${query}.ref.i2.blocks

PN_name_conv <- read.delim("all_on_T2T_ref_equivalences_filtered_no_copies_symbols_v1_vs_v5.csv", header = FALSE, quote = "\"", stringsAsFactors = FALSE)

# View the data
head(PN_name_conv)

anchors <- read.table("PN40024_T2T.ref.i2.anchors", header = FALSE, sep = "\t", stringsAsFactors = FALSE)
head(anchors)

anchors$V1 <- sub("\\_t\\d+$", "", anchors$V1)
# unique_gene_markers <- unique(all_markers)  # add as many vectors as needed

names(anchors)[names(anchors) == "V1"] <- "gene.ID"

DefenseGenes<-read.csv("Table6.csv")

DefenseGenes$Protein.ID <- sub("\\.t\\d+$", "", DefenseGenes$Protein.ID)
names(PN_name_conv)[names(PN_name_conv) == "V1"] <- "Protein.ID"
PN_name_conv <- PN_name_conv[PN_name_conv$V10 == "5.1", ]


DefenseGenes_v5<-merge(DefenseGenes, PN_name_conv, by="Protein.ID")
DefenseGenes_v5 <- DefenseGenes_v5[!duplicated(DefenseGenes_v5$Protein.ID), ]
names(DefenseGenes_v5)[names(DefenseGenes_v5) == "V2"] <- "gene.ID"

DefenseGenes_CS08<-merge(DefenseGenes_v5, anchors, by="gene.ID")

DefenseGenes_CS08 <- DefenseGenes_CS08[!duplicated(DefenseGenes_CS08$gene.ID), ]
names(DefenseGenes_CS08)[names(DefenseGenes_CS08) == "V2"] <- "gene.ID_CS08"

DefenseGenes_CS08$gene.ID_CS08 <- sub("\\.t\\d+$", "", DefenseGenes_CS08$gene.ID_CS08)


DefenseGenes_CS08_sub<- DefenseGenes_CS08[, c("Functional.category", "Functional.subcategory", "gene.ID_CS08")]
names(DefenseGenes_CS08_sub)[names(DefenseGenes_CS08_sub) == "gene.ID_CS08"] <- "gene_id"

```

