## Dataset origin and pre-processing

This dataset includes single-nuclei RNA-seq from 15 samples. Data was generated by Illumina Novaseq sequencing from 10x Genomics  Single Cell 3' end v4 (polyA) library prep kit.

The raw data were processed as described in steps 00 and 01 in this repository.

The starting file E. necator analysis is "./rds_outputs/Clean_pre-processed_seurat.rds" created in step 01.

## Obtain and inspect Seurat object 

```R
library(Seurat)
library(Matrix)
library(tidyverse)

#read the seurat object 
obj<-readRDS("./rds_outputs/Clean_pre-processed_seurat.rds")

#join layers
obj_joined <- JoinLayers(obj)

```

## Identify grapevine and fungal genes based on naming patterns
```R
# set the patterns
grapevine_pattern <- "VIT"
pathogen_pattern <- "gene-HI914"
Chlr_pattern <- "gene-ViviC"
Mit_pattern <- "gene-ViviM"

# Extract raw counts
counts <- GetAssayData(obj_joined, assay = "RNA", layer = "counts")

# Identify grapevine and pathogen genes
grapevine_genes <- grep(grapevine_pattern, rownames(counts), value = TRUE)
pathogen_genes  <- grep(pathogen_pattern, rownames(counts), value = TRUE)
Chlr_genes <- grep( Chlr_pattern, rownames(counts), value = TRUE)
Mit_genes <- grep(Mit_pattern, rownames(counts), value = TRUE)
```

## Calculate fungal statistics per nuclei

```R
# Calculate fungal metrics asn save in the metadata
meta <- obj_joined@meta.data
meta$nCount_fungal <- Matrix::colSums(counts[pathogen_genes, , drop = FALSE])
meta$nFeature_fungal <- Matrix::colSums(counts[pathogen_genes, , drop = FALSE] > 0)
meta$percent_fungal <- 100 * meta$nCount_fungal / meta$nCount_RNA
meta$fungal_score <- meta$percent_fungal * log1p(meta$nCount_fungal)
meta$percent_fungal_features <- 100 * meta$nFeature_fungal / meta$nFeature_RNA

# Add back to Seurat object
obj_joined <- AddMetaData(obj_joined, metadata = meta)
```


## Quantile selection of fungal nuclei 
#### Different quantile cutoffs of the proportion of fungal features were tested, but the 95th was used for the analysis

```R
library(viridis)

# filter to keep only nuclei with fungal reads
meta_fu <- meta%>%
   filter(percent_fungal_features > 0)

# Quantiles cutoffs on percent_fungal_features (Y-axis)
q_y <- quantile(meta_f$percent_fungal, probs = c(0.25, 0.5, 0.75, 0.9, 0.95,0.99), na.rm = TRUE)


hexbin <- ggplot(meta_fu, aes(x = nCount_fungal, y = percent_fungal_features)) +
   geom_point(bins = 500) +
   scale_fill_viridis_c() +
   geom_hline(yintercept = q_y, linetype = "dashed", color = "blue") +
   theme_minimal() +
   labs(
      x = "nCount_fungal",
      y = "% Fungal features",
      title = "Hexbin Plot with Quantiles",
      subtitle = "Blue = quantiles (25th, 50th, 75th, 90th, 95th, 99th)")

ggsave(plot=hexbin, filename = "plots/Hexbin_Plot_quantiles_pathogen_reads_nCount.pdf",
       width = 20, height = 17, units = "cm")


# Set 95th quantile cutoff on the proportion of fungal features
q95 <- quantile(meta_fu$percent_fungal_features, 0.95, na.rm = TRUE)

# Obtain the logical index to extract the nuclei with higher proportion of fungal features 
idx_95 <- meta$percent_fungal_features > q95

# Subset nuclei based on cutoff
j_obj_95 <- subset(obj_joined, cells = rownames(meta)[idx_95])

# Check number of nuclei 
table(j_obj_95$Treatment)
# Inf_1DPI Inf_5DPI 95th
# 306     3517
```


## Outlier nuclei filtering
#### Some nuclei had an abnormally large number of reads mapping to fungal features (compared to the rest of the nuclei at a similar proportion of fungal features). We decide to filter them out to reduce noise.

```R
# set working objects
obj <- j_obj_95
meta <- obj@meta.data

# Fit the LOESS model and calculate residual
loess_fit <- loess(nCount_fungal ~ percent_fungal_features, data = meta)
meta$fitted <- predict(loess_fit)
meta$residual <- meta$nCount_fungal - meta$fitted
   
#Calculate robust Z-scores of residuals Median Absolute Deviation (MAD scaling)
mad_res <- mad(meta$residual)
med_res <- median(meta$residual)
meta$z_residual <- abs(meta$residual - med_res) / mad_res

# Flag nuceli based on the robust Z-score threshold (|Z| > 2.5) 
meta$keep <- meta$z_residual < 2.5

# Plot residuals with marked outliers
p <- ggplot(meta, aes(x = percent_fungal_features, y = nCount_fungal)) +
  geom_hex(bins = 300) +
  geom_smooth(method = "loess", se = FALSE, color = "blue", linewidth = 0.6) +
  geom_point(data = subset(meta, keep == FALSE),
             aes(x = percent_fungal_features, y = nCount_fungal),
             color = "red", size = 0.1, alpha = 0.6) +
  theme_minimal() +
  labs(title = "Z < 2.5",
       subtitle = paste("Residual threshold:", round(med_res + 2.5 * mad_res)),
       x = "% Fungal Features",
       y = "nCount_fungal")

ggsave(filename = "plots/95th_hexbin_filtered_Z2.5_z_thresh.pdf",
          plot = p, width = 8, height = 8, units = "cm")

   
# Update metadata
obj@meta.data <- meta

# Subset the Seurat object keep only nuclei passing the filter
cells_to_keep <- rownames(meta[meta$keep, ])
j_obj_95_outliers_Z2.5 <- subset(obj, cells = cells_to_keep) 


# Check final filtered dataset
table(j_obj_95_outliers_Z2.5$Treatment)


```


